src/dotenv/cli.py

```diff
@@ -19,19 +19,27 @@
 
 @click.group()
 @click.option('-f', '--file', default=os.path.join(os.getcwd(), '.env'),
-              type=click.Path(exists=True),
+              type=click.Path(file_okay=True),
               help="Location of the .env file, defaults to .env file in current working directory.")
 @click.option('-q', '--quote', default='always',
               type=click.Choice(['always', 'never', 'auto']),
               help="Whether to quote or not the variable values. Default mode is always. This does not affect parsing.")
+@click.option('-e', '--export', default=False,
+              type=click.BOOL,
+              help="Whether to write the dot file as an executable bash script.")
 @click.version_option(version=__version__)
 @click.pass_context
-def cli(ctx, file, quote):
+def cli(ctx, file, quote, export):
     # type: (click.Context, Any, Any) -> None
     '''This script is used to set, get or unset values from a .env file.'''
     ctx.obj = {}
-    ctx.obj['FILE'] = file
     ctx.obj['QUOTE'] = quote
+    ctx.obj['EXPORT'] = ctx.obj.get('EXPORT', False)
+    if ctx.obj['EXPORT']:
+        name, extension = os.path.splitext(file)
+        if not extension:
+            file = name+".sh"
```

Comment:
It seemed to me that the purpose of adding **export** to the file, was so that the file was also a valid shell script. It didn't seem obvious to me why someone would want to have **export**, but would exclusively dislike having the extension; I thought it would save the user from having to take that extra step and rename it themselves.

Reply:
Thank you for suggesting the tests by the way. I found and fixed a bug on line 37 when testing to see if **export** was used in the file.